src/GslCore/DnaCreation.fs

```diff
@@ -52,63 +52,58 @@ let adjustToPhysical (feat:sgd.Feature) (f:RelPos) =
 /// Transform all non gXXX forms of gene into gXX forms.
 let translateGenePrefix (pragmas:PragmaCollection) (gd : GenomeDef) (gPart : StandardSlice) =
     match gPart with
-    | PROMOTER ->   
-        { left = { x = match pragmas.TryFind "promlen"  with
-                        | None -> -gd.getPromLen() 
-                        | Some p -> p.args.[0] |> int |> (*)-1<OneOffset> ;  
-                   relTo = FivePrime
-                 };
+    | PROMOTER ->
+        let leftPos =
+            match pragmas.TryGetOne "promlen" with
+            | None -> -gd.getPromLen() 
+            | Some p -> (int p) * -1<OneOffset> 
+        {left = {x = leftPos; relTo = FivePrime};
          lApprox = true;
          rApprox = false;
-         right = { x = -1<OneOffset>; relTo = FivePrime } 
-        }
+         right = {x = -1<OneOffset>; relTo = FivePrime}}
     | UPSTREAM ->
-        {left = {x = -gd.getFlank(); relTo = FivePrime};
-         lApprox = true;
-         rApprox = false;
-         right = { x = -1<OneOffset>; relTo = FivePrime } }
+       {left = {x = -gd.getFlank(); relTo = FivePrime};
+        lApprox = true;
+        rApprox = false;
+        right = {x = -1<OneOffset>; relTo = FivePrime}}
     | TERMINATOR ->
+        let rightPos =
+            match pragmas.TryGetOne "termlen" with
+            | None -> gd.getTermLen() 
+            | Some p -> (int p) * 1<OneOffset> 
         {left = {x = 1<OneOffset>; relTo = ThreePrime};
          lApprox = false;
          rApprox = true;
-         right = { x = match pragmas.TryFind "termlen"  with
-                        | None -> gd.getTermLen() 
-                        | Some p -> p.args.[0] |> int |> (*)1<OneOffset> ; 
-                   relTo = ThreePrime 
-                  }
-        }
+         right = {x = rightPos; relTo = ThreePrime}}

```

Comment:
This change is eliding part of the custom terminator length support.  I think simply avoiding this change fixes the problem.

Reply:
What is the problem?